src/main/java/com/redislabs/redisgraph/impl/api/ContextedRedisGraph.java

```diff
@@ -59,6 +59,27 @@ protected ResultSet sendQuery(String graphId, String preparedQuery) {
         }
     }
 
+    /**
+     * Sends the read-only query over the instance only connection
+     * @param graphId graph to be queried
+     * @param preparedQuery prepared query
+     * @return Result set with the query answer
+     */
+    @Override
+    protected ResultSet sendReadOnlyQuery(String graphId, String preparedQuery) {
+        Jedis conn = getConnection();
+        try {
+            List<Object> rawResponse = (List<Object>) conn.sendCommand(RedisGraphCommand.RO_QUERY, graphId, preparedQuery, Utils.COMPACT_STRING);
+            return new ResultSetImpl(rawResponse, this, caches.getGraphCache(graphId));
+        }
+        catch (JRedisGraphRunTimeException rt) {
+            throw rt;
+        }
+        catch (JedisDataException j) {
+            throw new JRedisGraphCompileTimeException(j);
+        }
```

Comment:
Two of these error classes were removed. Now there is only `JRedisGraphException`.